File: PRACTICA AUXI/java/INTRODUCCION A LA PROGRAMACION/Coche.java
/// "NOMBRE: QUENTA MAMANI NICK RANDY"
/// "CI: 10037571"
///
/// "3. Crea una clase Coche con marca, modelo y velocidad"
/// "a) Agrega un método acelerar () que aumente la velocidad en 10"
/// "b) Agrega un método frenar () que disminuya la velocidad en 5"
/// "c) Crea dos coches, aceléralos, frénalos y muestra sus velocidades"
///
public class Coche {
    private String marca;
    private String modelo;
    private int velocidad;

    public Coche(String marca, String modelo) {
        this.marca = marca;
        this.modelo = modelo;
        this.velocidad = 0;
    }

    public void acelerar() {
        velocidad += 10;
    }

    public void frenar() {
        velocidad -= 5;
    }

    public String mostrarVelocidad() {
        return marca + " " + modelo + " tiene una velocidad de " + velocidad + " km/h.";
    }

    public static void main(String[] args) {
        Coche auto1 = new Coche("Toyota", "Supra");
        Coche auto2 = new Coche("Nissan", "skyline gtr r34");

        auto1.acelerar();
        auto2.acelerar();

        auto1.frenar();
        auto2.frenar();

        System.out.println(auto1.mostrarVelocidad());
        System.out.println(auto2.mostrarVelocidad());
    }
}

File: PRACTICA AUXI/python/INTRODUCCION A LA PROGRAMACION/Coche.py
"NOMBRE: QUENTA MAMANI NICK RANDY"
"CI: 10037571"

"3. Crea una clase Coche con marca, modelo y velocidad"
"a) Agrega un método acelerar () que aumente la velocidad en 10"
"b) Agrega un método frenar () que disminuya la velocidad en 5"
"c) Crea dos coches, aceléralos, frénalos y muestra sus velocidades"


class Coche:
    def __init__(self, marca, modelo, velocidad=0):
        self.marca = marca
        self.modelo = modelo
        self.velocidad = velocidad

    def acelerar(self):
        self.velocidad += 10

    def frenar(self):
        self.velocidad -= 5

    def mostrar_velocidad(self):
        return f"{self.marca} {self.modelo} tiene una velocidad de {self.velocidad} km/h."


if __name__ == "__main__":
    auto1 = Coche("Toyota", "Supra")
    auto2 = Coche("Nissan", "skyline gtr r34")

    auto1.acelerar()
    auto2.acelerar()

    auto1.frenar()
    auto2.frenar()

    print(auto1.mostrar_velocidad())
    print(auto2.mostrar_velocidad())
